policyeval/base_worker.go

```diff
@@ -177,23 +175,71 @@ func (w *BaseWorker) handlePolicy(ctx context.Context, eval *sdk.ScalingEvaluati
 			continue
 		}
 
-		winningAction = sdk.PreemptScalingAction(winningAction, action)
-		if winningAction == action {
-			winningHandler = checkHandler
+		group := checkEval.Check.Group
+		checkGroups[group] = append(checkGroups[group], checkResult{
+			action:  action,
+			handler: checkHandler,
+		})
+	}
+
+	// winner is the final check that will be executed after the check groups
+	// are processed.
+	var winner checkResult
+
+	for group, results := range checkGroups {
+		// Decide which action wins in the group. The decision processes still
+		// picks the safest choice, but it handles `none` actions a little
+		// differently.
```

Comment:
I'm not super familiar with how the `sdk.PreemptScalingAction` method works, but should we be enforcing non-overlapping bounds for checks within a group?